client/cmd_noauth.go

```diff
@@ -98,6 +99,14 @@ func (c *Client) Authenticate(auth sasl.Client) error {
 		Mechanism: mech,
 	}
 
+	irOk, err := c.Support("SASL-IR")
+	if err != nil {
+		return err
+	}
+	if irOk {
+		cmd.InitialResponse = base64.StdEncoding.EncodeToString(ir)
```

Comment:
We should probably unset `ir` here, so that it's not duplicated in `responses.Authenticate`?

Reply:
>responses.Authenticate

You mean commands.Authenticate, right?

Reply:
What is duplicated tho? I'm not sure if this still applies after two commits I just pushed.

Reply:
I'm concerned about `ir` being set for both the command and the response:

```go
	if irOk {
		cmd.InitialResponse = ir // set here
	}

	res := &responses.Authenticate{
		Mechanism:       auth,
		InitialResponse: ir, // set here
		RepliesCh:       make(chan []byte, 10),
	}
```

If we set it for the command, we shouldn't need to set it for the response?

Reply:
Ah, I see. f22f246.